File: memory-domain/src/main/java/com/memory/dto/search/SearchResultResponse.java
package com.memory.dto.search;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Builder
public class SearchResultResponse {

    private List<MemorySearchResponse> memories;
    private PageInfo pageInfo;
    private SearchMetadata metadata;

    @Getter
    @Builder
    public static class PageInfo {
        private int currentPage;
        private int totalPages;
        private int pageSize;
        private long totalElements;
        private boolean hasNext;
        private boolean hasPrevious;
    }

    @Getter
    @Builder
    public static class SearchMetadata {
        private SearchType searchType;
        private String query;
        private List<String> hashtags;
        private LocalDate fromDate;
        private LocalDate toDate;
        private long searchTimeMs;
    }
}

File: memory-domain/src/main/kotlin/com/memory/dto/search/SearchResultResponse.kt
package com.memory.dto.search

import java.time.LocalDate

data class SearchResultResponse(
    val memories: List<MemorySearchResponse>,
    val pageInfo: PageInfo,
    val metadata: SearchMetadata
) {
    data class PageInfo(
        val currentPage: Int,
        val totalPages: Int,
        val pageSize: Int,
        val totalElements: Long,
        val hasNext: Boolean,
        val hasPrevious: Boolean
    )

    data class SearchMetadata(
        val searchType: SearchType,
        val query: String?,
        val hashtags: List<String>?,
        val fromDate: LocalDate?,
        val toDate: LocalDate?,
        val searchTimeMs: Long
    )
}
